src/main/java/org/opensrp/web/rest/ImportBulkDataController.java

```diff
@@ -0,0 +1,115 @@
+	}
+
+	@PostMapping(headers = { "Accept=multipart/form-data" }, produces = {
+			MediaType.APPLICATION_JSON_VALUE }, value = "/organizations")
+	public ResponseEntity importOrganizationsData(@RequestParam("file") MultipartFile file)
+			throws IOException {
+
+		List<Map<String, String>> csvClients = readCSVFile(file);
+		CsvBulkImportDataSummary csvBulkImportDataSummary = importBulkDataService
+				.convertandPersistPractitionerdata(csvClients);
+
+		generateCSV(csvBulkImportDataSummary);
+		File csvFile = new File(SAMPLE_CSV_FILE);
+
+		return ResponseEntity.ok()
+				.header("Content-Disposition", "attachment; filename=" + "importsummaryreport" + ".csv")
+				.contentLength(csvFile.length())
+				.contentType(MediaType.parseMediaType("text/csv"))
+				.body(new FileSystemResource(csvFile));
+	}
+
+	@PostMapping(headers = { "Accept=multipart/form-data" }, produces = {
+			MediaType.APPLICATION_JSON_VALUE }, value = "/practitioners")
+	public ResponseEntity importPractitionersData(@RequestParam("file") MultipartFile file) throws
+			IOException {
+
+		List<Map<String, String>> csvClients = readCSVFile(file);
+		CsvBulkImportDataSummary csvBulkImportDataSummary = importBulkDataService
+				.convertandPersistPractitionerdata(csvClients);
+
+		generateCSV(csvBulkImportDataSummary);
+		File csvFile = new File(SAMPLE_CSV_FILE);
+
+		return ResponseEntity.ok()
+				.header("Content-Disposition", "attachment; filename=" + "importsummaryreport" + ".csv")
+				.contentLength(csvFile.length())
+				.contentType(MediaType.parseMediaType("text/csv"))
+				.body(new FileSystemResource(csvFile));
+	}
+
+	private List<Map<String, String>> readCSVFile(MultipartFile file) throws IOException {
+		List<Map<String, String>> csvClients = new ArrayList<>();
+		try (Reader reader = new InputStreamReader(file.getInputStream());
+		     CSVParser parser = new CSVParser(reader, CSVFormat.DEFAULT.withHeader())) {
+
+			List<CSVRecord> records = parser.getRecords();
+			for (CSVRecord record : records) {
+				csvClients.add(record.toMap());
+			}
+		}
+		return csvClients;
+	}
+
+	private void generateCSV(CsvBulkImportDataSummary csvBulkImportDataSummary) {
+		try (
+				BufferedWriter writer = Files.newBufferedWriter(Paths.get(SAMPLE_CSV_FILE));
+				CSVPrinter csvPrinter = new CSVPrinter(writer, CSVFormat.DEFAULT);
+		) {
+			csvPrinter.printRecord("Total Number of Rows in the CSV ", csvBulkImportDataSummary.getNumberOfCsvRows());
+			csvPrinter.printRecord("Rows processed ", csvBulkImportDataSummary.getNumberOfRowsProcessed());
+			csvPrinter.printRecord("\n");
+
+			csvPrinter.printRecord("Row Number", "Reason of Failure");
+			for (FailedRecordSummary failedRecordSummary : csvBulkImportDataSummary.getFailedRecordSummaryList()) {
+				csvPrinter.printRecord(failedRecordSummary.getRowNumber(), failedRecordSummary.getReasonOfFailure());
+			}
+			csvPrinter.flush();
+		}
+		catch (Exception e) {
+			logger.error("Failed to generate CSV " + e.getMessage());
```

Comment:
add e as second param

Reply:
if this fails the API will return a 200 code even though the data returned may not be correct. Remove this catch block so that an error is returned on the API. Alternatively re throw a custom exception